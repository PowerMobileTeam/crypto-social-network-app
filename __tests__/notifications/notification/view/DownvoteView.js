import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';
import DownvoteView from '../../../../src/notifications/notification/view/DownvoteView';
import styles from '../../../../src/notifications/notification/style';

// fake data generation
import boostNotificationFactory from '../../../../__mocks__/fake/notifications/BoostNotificationFactory';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

/**
 * For activity
 */
it('renders correctly for activity', () => {

  const entity = boostNotificationFactory('downvote', 'activity');

  const notification = renderer.create(
    <DownvoteView styles={styles} entity={entity}/>
  ).toJSON();

  expect(notification).toMatchSnapshot();
});

/**
 * For comment
 */
it('renders correctly for comment', () => {

  const entity = boostNotificationFactory('downvote', 'comment');

  const notification = renderer.create(
    <DownvoteView styles={styles} entity={entity}/>
  ).toJSON();

  expect(notification).toMatchSnapshot();
});

/**
 * For object
 */
it('renders correctly for object', () => {

  const entity = boostNotificationFactory('downvote', 'object');

  const notification = renderer.create(
    <DownvoteView styles={styles} entity={entity}/>
  ).toJSON();

  expect(notification).toMatchSnapshot();
});