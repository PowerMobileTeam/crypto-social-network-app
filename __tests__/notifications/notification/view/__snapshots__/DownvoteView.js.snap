// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for activity 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#444",
          "fontWeight": "bold",
        }
      }
    >
      someUser
    </Text>
     
    down voted
     
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#444",
          "fontWeight": "bold",
        }
      }
    >
      some title
    </Text>
  </Text>
</View>
`;

exports[`renders correctly for comment 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#444",
          "fontWeight": "bold",
        }
      }
    >
      someUser
    </Text>
     
    down voted
     
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#444",
          "fontWeight": "bold",
        }
      }
    >
       your comment 
    </Text>
  </Text>
</View>
`;

exports[`renders correctly for object 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#444",
          "fontWeight": "bold",
        }
      }
    >
      someUser
    </Text>
     
    down voted
     
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#444",
          "fontWeight": "bold",
        }
      }
    >
      some title
    </Text>
  </Text>
</View>
`;
